components/feature/customtabs/src/test/java/mozilla/components/feature/customtabs/CustomTabIntentProcessorTest.kt

```diff
@@ -72,9 +76,52 @@ class CustomTabIntentProcessorTest {
         assertNotNull(customTabSession.customTabConfig)
     }
 
+    @Test
+    fun doNotProcessOtherIntent() = runBlockingTest {
+        val sessionManager: SessionManager = mock()
+        val useCase: SessionUseCases.DefaultLoadUrlUseCase = mock()
+
+        val handler = CustomTabIntentProcessor(sessionManager, useCase, testContext.resources)
+
+        val intent = mock<Intent>()
+        whenever(intent.action).thenReturn(Intent.ACTION_BATTERY_LOW)
+        whenever(intent.hasExtra(CustomTabsIntent.EXTRA_SESSION)).thenReturn(true)
+        whenever(intent.dataString).thenReturn("http://mozilla.org")
+        whenever(intent.putExtra(any<String>(), any<String>())).thenReturn(intent)
+
+        assertFalse(handler.process(intent))
+        verify(sessionManager, never()).add(anySession(), eq(false), eq(null), eq(null))
+        verify(intent, never()).putExtra(eq(EXTRA_SESSION_ID), any<String>())
+    }
+
+    @Test
+    fun doNotProcessCustomTabIntentWhenBrowserUiIsSet() = runBlockingTest {
+        val sessionManager: SessionManager = mock()
+        val useCase: SessionUseCases.DefaultLoadUrlUseCase = mock()
+
+        val handler = CustomTabIntentProcessor(sessionManager, useCase, testContext.resources)
+
+        val intent = mock<Intent>()
+        whenever(intent.action).thenReturn(Intent.ACTION_VIEW)
+        whenever(intent.hasExtra(CustomTabsIntent.EXTRA_SESSION)).thenReturn(true)
+        whenever(intent.dataString).thenReturn("http://mozilla.org")
+        whenever(intent.getBooleanExtra(EXTRA_USER_OPT_OUT_FROM_CUSTOM_TABS, false)).thenReturn(true)
+        whenever(intent.flags).thenReturn(FLAG_ACTIVITY_NEW_TASK)
+        whenever(intent.putExtra(any<String>(), any<String>())).thenReturn(intent)
+
+        assertFalse(handler.process(intent))
+        verify(sessionManager, never()).add(anySession(), eq(false), eq(null), eq(null))
+        verify(intent, never()).putExtra(eq(EXTRA_SESSION_ID), any<String>())
+    }
+
     @Suppress("UNCHECKED_CAST")
     private fun <T> anySession(): T {
         any<T>()
         return null as T
     }
+
+    companion object {
+        private const val EXTRA_USER_OPT_OUT_FROM_CUSTOM_TABS =
+            "android.support.customtabs.extra.user_opt_out"
+    }
```

Comment:
It's somehow weird that this is the only `EXTRA_*` constant in `CustomTabsIntent` that is private. Should we file a bug to make that public? 🤔